tox.ini

```
@@ -77,8 +77,23 @@ deps =
     pytest-cov
 
 [testenv:func]
+setenv =
+  TEST_JUJU_CHANNEL = 2.9/stable
 changedir = {toxinidir}/tests/functional
 commands = functest-run-suite {posargs:--keep-faulty-model}
 deps =
   git+https://github.com/openstack-charmers/zaza.git#egg=zaza
   -r {toxinidir}/tests/functional/requirements.txt
+
+[testenv:func31]
+setenv =
+  TEST_JUJU_CHANNEL = 3.1/stable
+changedir = {toxinidir}/tests/functional
+commands =
+  # NOTE(mertkirpici): functest-run-suite -b jammy focal
+  # or functest-run-suite -b jammy -b focal does not work
+  functest-run-suite --keep-faulty-model -b jammy
+  functest-run-suite --keep-faulty-model -b focal
```

```suggestion
  functest-run-suite --bundle jammy focal {posargs:--keep-faulty-model}
```
or maybe we can take advantage of aliases in zaza, so `juju_30: jammy` can be used so it's more obvious that which bundles are run on juju3.1.

---

+1 on adding back `postargs`

---

```suggestion
  functest-run-suite -b jammy {posargs:--keep-faulty-model}
  functest-run-suite -b focal {posargs:--keep-faulty-model}
```